index.d.ts

```diff
@@ -0,0 +1,20 @@
+declare module 'apollo-datasource-mongodb' {
+  import { DataSource, DataSourceConfig } from 'apollo-datasource'
+  import * as mongoose from 'mongoose'
+
+  export {
+    DataSourceConfig
+  }
+
+  interface Options { ttl: number }
+
+  export abstract class MongoDataSource<T extends mongoose.Document, TContext = any> extends DataSource {
```

Comment:
```suggestion
  export abstract class MongoDataSource<T extends mongoose.Document, TContext = any> extends DataSource<TContext> {
```

`DataSource` is also a generic type, which optionally takes in the context type, which is what we should do imo.

Reply:
Also, I don't think the `collection` type makes these typings compatible with the native mongodb connector, which has a generic type for collection